Shared/alerts/AlertConditionsController.cpp

```diff
@@ -1058,6 +1070,17 @@ void AlertConditionsController::addStoredConditions()
   });
 }
 
+bool AlertConditionsController::conditionAlreadyAdded(const QString& conditionName)
+{
+  for (auto i = 0; i < m_conditions->rowCount(); i++)
+  {
+    const auto* condition = m_conditions->conditionAt(i);
+    if (condition->name() == conditionName)
+      return true;
+  }
+  return false;
+}
```

Comment:
Is it possible to use a range loop? If not, this is fine.

Reply:
This class itself (subclass of QAbstractItemModel) does not directly support ranged loop. But we can add the following to the header to expose the underlying QList<AlertConditions*> which is the root object really. Not many examples in the codebase where we define the body in the header file. This seems simple enough to just put in the header though.

```
QList<AlertCondition*>::iterator begin() { return m_conditions.begin(); }
QList<AlertCondition*>::iterator end() { return m_conditions.end(); }
const QList<AlertCondition*>::const_iterator const_begin() { return m_conditions.constBegin(); }
const QList<AlertCondition*>::const_iterator const_end() { return m_conditions.constEnd(); }
```

Reply:
Let's see if this becomes a persistent issue. Short of that I don't think it's worth solving in isolation for this one case.